### PR TITLE
Search Component

### PR DESCRIPTION
**Pull request by phillipahereza**
## Description

Implements Search Component

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

screen shot 2019-03-01 at 9 58 25 am

screen shot 2019-03-01 at 9 58 11 am

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: dominion/dominion_ui/public/index.html [hunk 1]
```diff
@@ -55,6 +55,11 @@
       })(window.location);
     </script>
     <!-- end single page apps for github pages -->
+
+    <!-- test purposes -->
+    <script type="text/javascript">
+        var MAPIT = { url: "https://mapit.hurumap.org" };
+    </script>
```
Comment (KhadijaMahanga) [inline dominion/dominion_ui/public/index.html:62]: Shouldn't this be injected from the server (`dominion`)  and not here?

**Comment by karimkawambwa** [comment]
@kilemensi @KhadijaMahanga 👀

File: dominion/dominion_ui/public/index.html [hunk 2]
```diff
@@ -55,6 +55,11 @@
       })(window.location);
     </script>
     <!-- end single page apps for github pages -->
+
+    <!-- test purposes -->
+    <script type="text/javascript">
+        var MAPIT = { url: "https://mapit.hurumap.org" };
```
Comment (kilemensi) [inline dominion/dominion_ui/public/index.html:61]: This is available from `settings.py` i.e. `HURUMAP['mapit']['url']`

File: dominion/dominion_ui/src/components/Search/SearchResults.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const maxResults = 6;
+
+function SearchResults({ classes, results }) {
+  return (
+    <div className={classes.root}>
+      <Grid container direction="row" justify="flex-end">
+        <List className={classes.list}>
+          {results.slice(0, maxResults).map(result => (
+            <ListItem key={weakKey(result)} className={classes.listItem} button>
```
Comment (kilemensi) [inline dominion/dominion_ui/src/components/Search/SearchResults.js]: Unless I'm mistaken, each region have a name and code which are unique. We should use them for key. (No need for extra library).

File: dominion/templates/_base.html [hunk 4]
```diff
@@ -4,6 +4,10 @@
 {% block head_css %}
 {{ block.super }}
 <link rel="stylesheet" href="{% sass_src 'css/dominion.scss' %}">
+<link
+    href="https://fonts.googleapis.com/css?family=Lora:400,700|Montserrat:200,300,400,500,600,700,800,900"
+    rel="stylesheet"
+/>
```
Comment (kilemensi) [inline dominion/templates/_base.html:10]: Nice catch, :+1:

File: dominion/dominion_ui/src/components/Hero/index.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    backgroundRepeat: 'no-repeat',
+    backgroundSize: 'cover'
+  }
+});
+
+function HomeHero({ classes }) {
+  return (
+    <Grid container item xs={12} className={classes.root}>
+      <Hero />
+    </Grid>
```
Comment (kilemensi) [inline dominion/dominion_ui/src/components/Hero/index.js]: What does `Grid` actually gives us here as opposed to just returning `<Hero />`?

File: dominion/dominion_ui/src/components/Header/Search.js [hunk 6]
```diff
@@ -1,65 +1,138 @@
...
+              close={closeModal}
+            >
+              <SearchModalContent />
+            </Modal>
+            {children}
+          </React.Fragment>
+        )}
+      </React.Fragment>
+    );
+  }
```
Comment (kilemensi) [inline dominion/dominion_ui/src/components/Header/Search.js:129]: I like the way you're breaking components down but *so*.*many*.`<Grid container ...`. Wouldn't it be better to have a single `<Grid ...` on the `return (`? After all, the point of a grid is to layout **multiple** components.

**Review by kilemensi — COMMENTED** [review]
LGTM @karimkawambwa :+1:

----------------------------------------------

A few general observations though (and this is across the whole code base, not just this PR).

 1. Lets read the docs on the Material UI [Grid](https://material-ui.com/layout/grid/) and make sure we use it: **when** needed and in the way it's been _designed_ to be used e.g. No need to start with `<Grid item`, almost always you start with `<Grid container` to layout **multiple** components. But you can have a grid item _inside_ a grid container that also acts as a container for its children i.e.
```js
   <Grid container ...
      <Grid item container ...
```
 2. Mobile first design: Always start by writing styles that would work in mobile and the customize for large screens using `[theme.breakpoints.up(...`. Doing it the other way i.e.`[theme.breakpoints.down(...` makes it quite difficult to follow how the UI flows as the screen size increases,
 3. Lets use relative units `rem` (and `em` where appropriate) rather than fixed `px`, and
 4. Lets pick one style and stick to when exporting components i.e. order of  `withRoot`, `withWidth` and `withStyle`. I prefer `withWidth()(withStyles(styles)(Component));` but whatever we decide, lets stick to rather than mixing and matching. This will enable us to quickly spot when/where we miss something.

cc @KhadijaMahanga, @nerdlet

File: dominion/dominion_ui/public/index.html (hunk 2 above)
Comment (karimkawambwa) [reply dominion/dominion_ui/public/index.html:61]: This is purely for when I ran react with `yarn start`

File: dominion/dominion_ui/src/components/Hero/index.js (hunk 5 above)
Comment (karimkawambwa) [reply dominion/dominion_ui/src/components/Hero/index.js]: I will clean this up, this was  method phillip was doing i went a different direction and forgot to clean this up.

File: dominion/dominion_ui/src/components/Header/Search.js (hunk 6 above)
Comment (karimkawambwa) [reply dominion/dominion_ui/src/components/Header/Search.js:129]: Many of these components have children that need their own layout.
I agree i might have over used some places, I think i will clean that up  and keep this in mind for next sprint.

**Review by KhadijaMahanga — COMMENTED** [review]
> LGTM @karimkawambwa 
> 
> A few general observations though (and this is across the whole code base, not just this PR).
> 
> 1. Lets read the docs on the Material UI [Grid](https://material-ui.com/layout/grid/) and make sure we use it: **when** needed and in the way it's been _designed_ to be used e.g. No need to start with `<Grid item`, almost always you start with `<Grid container` to layout **multiple** components. But you can have a grid item _inside_ a grid container that also acts as a container for its children i.e.
> 
> ```js
>    <Grid container ...
>       <Grid item container ...
> ```
> 
> 1. Mobile first design: Always start by writing styles that would work in mobile and the customize for large screens using `[theme.breakpoints.up(...`. Doing it the other way i.e.`[theme.breakpoints.down(...` makes it quite difficult to follow how the UI flows as the screen size increases,
> 2. Lets use relative units `rem` (and `em` where appropriate) rather than fixed `px`, and
> 3. Lets pick one style and stick to when exporting components i.e. order of  `withRoot`, `withWidth` and `withStyle`. I prefer `withWidth()(withStyles(styles)(Component));` but whatever we decide, lets stick to rather than mixing and matching. This will enable us to quickly spot when/where we miss something.
> 
> cc @KhadijaMahanga, @nerdlet

Noted

**Pull request merged**